library/src/c_api.rs

```diff
@@ -155,13 +157,20 @@ pub extern "C" fn shorebird_free_string(c_string: *mut c_char) {
 /// Check for an update.  Returns true if an update is available.
 #[no_mangle]
 pub extern "C" fn shorebird_check_for_update() -> bool {
-    return updater::check_for_update();
+    log_on_error(updater::check_for_update, "checking for update", false)
 }
 
 /// Synchronously download an update if one is available.
 #[no_mangle]
 pub extern "C" fn shorebird_update() {
-    updater::update();
+    log_on_error(
+        || {
+            updater::update()?;
```

Comment:
This closure isn't necessary.

Reply:
Actually it is, but another way would be to catch the update status, match it and log from the match.  (Or just log the update status and assume it has a Display property.

Reply:
Fixed.